core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -1117,7 +1124,14 @@ private[spark] class TaskSchedulerImpl(
       }
     }
 
-    executorsPendingDecommission.remove(executorId)
+    val removedDecomState = executorsPendingDecommission.remove(executorId)
+    removedDecomState.foreach(
+      s => {
+        removedExecutorsDueToDecom.put(executorId, s)
+        if (removedExecutorsDueToDecom.size >= REMOVED_EXECUTORS_MAX_SIZE) {
+          removedExecutorsDueToDecom.remove(removedExecutorsDueToDecom.head._1)
+        }
+      })
```

Comment:
```suggestion
    executorsPendingDecommission.remove(executorId).foreach(removedExecutorsDueToDecom.put(executorId, _))
```

Change to the Cache (above) will take care of eviction, etc.